src/deroff.rs

```diff
@@ -538,12 +538,311 @@ impl Deroffer {
+                    // self.skip_char(1);
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                }
+                return true;
+            // } else if self.prch(2) {
+            } else if self.prch(2) {
+                // self.skip_char(3);
+                self.s = self.skip_char(&self.s, Some(3)).to_owned();
+                return true;
+            } else {
+                return false;
+            }
+        } else if s0s1 == "\\*" {
+            let mut reg = String::new();
+            // if self.str_at(2) == "(" && self.prch(3) && self.prch(4) {
+            if Self::str_at(&self.s, 2) == "(" && self.prch(3) && self.prch(4) {
+                reg = self.s[3..5].to_owned();
+                // self.skip_char(5);
+                self.s = self.skip_char(&self.s, Some(5)).to_owned();
+            // } else if self.str_at(2) == "[" && self.prch(3) {
+            } else if Self::str_at(&self.s, 2) == "[" && self.prch(3) {
+                // self.skip_char(3);
+                self.s = self.skip_char(&self.s, Some(3)).to_owned();
+                // while !self.str_at(0).is_empty() && self.str_at(0) != "]" {
+                while !Self::str_at(&self.s, 0).is_empty() && Self::str_at(&self.s, 0) != "]" {
+                    // reg.push_str(self.str_at(0));
+                    reg.push_str(Self::str_at(&self.s, 0));
+                    // self.skip_char(1);
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                }
+                if let Some("]") = self.s.get(0..1) {
+                    // self.skip_char(1);
+                    self.s = self.skip_char(&self.s, None).to_owned();
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+
+            if self.reg_table.contains_key(&reg) {
+                // This unwrap is safe because of the if
+                self.s = self.reg_table.get(&reg).unwrap().to_owned();
+                // This is unimplemented, so tests fail:
+                // self.text_arg();
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    fn digit(&self, _: usize) -> bool {
+        unimplemented!()
+    }
+
+    fn size(&mut self) -> bool {
+        /* # We require that the string starts with \s */
+        if self.digit(2) || ("-+".contains(Self::str_at(&self.s, 2)) && self.digit(3)) {
+            // self.skip_char(3);
+            self.s = self.skip_char(&self.s, Some(3)).to_owned();
+            while self.digit(0) {
+                // self.skip_char(1);
+                self.s = self.skip_char(&self.s, None).to_owned();
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    fn condputs(&self, _: &str) {
+        unimplemented!()
+    }
```

Comment:
This also have been implemented.
https://github.com/rust-dc/fish-manpage-completions/blob/2e8c847b2bf0f04556126668aeef837425c34506/src/deroff.rs#L711-L722

Reply:
Fixed